sdk/core/src/headers/mod.rs

```diff
@@ -1,7 +1,24 @@
 mod utilities;
 
+use http::request::Builder;
 pub use utilities::*;
 
+#[must_use]
+pub fn add_optional_header<T: crate::AddAsHeader>(
+    item: &Option<T>,
+    mut builder: Builder,
+) -> Builder {
+    if let Some(item) = item {
+        builder = item.add_as_header(builder);
+    }
+    builder
+}
+
+#[must_use]
+pub fn add_mandatory_header<T: crate::AddAsHeader>(item: &T, builder: Builder) -> Builder {
```

Comment:
We should switch Cosmos to use these methods, but of course we can do that in another PR.

Reply:
It's included in this PR 👍